src/coordinator/ts/m3db/block/block_conversion.go

```diff
@@ -31,45 +34,81 @@ var (
 )
 
 // SeriesBlockToMultiSeriesBlocks converts M3DB blocks to multi series blocks
-func SeriesBlockToMultiSeriesBlocks(multiNamespaceSeriesList []MultiNamespaceSeries, seriesIteratorsPool encoding.MutableSeriesIteratorsPool) (MultiSeriesBlocks, error) {
+func SeriesBlockToMultiSeriesBlocks(multiNamespaceSeriesList []MultiNamespaceSeries, seriesIteratorsPool encoding.MutableSeriesIteratorsPool, stepSize time.Duration) (MultiSeriesBlocks, error) {
 	// todo(braskin): validate blocks size and aligment per namespace before creating []MultiNamespaceSeries
-	var multiSeriesBlocks MultiSeriesBlocks
-	for multiNamespaceSeriesIdx, multiNamespaceSeries := range multiNamespaceSeriesList {
-		consolidatedSeriesBlocks, err := newConsolidatedSeriesBlocks(multiNamespaceSeries, seriesIteratorsPool)
+	var (
+		multiSeriesBlocks MultiSeriesBlocks
+
+		commonTags      = make(map[string]string)
+		firstSeriesTags = make(map[string]string)
+	)
+
+	for seriesIdx, multiNamespaceSeries := range multiNamespaceSeriesList {
+		consolidatedSeriesBlocks, err := newConsolidatedSeriesBlocks(multiNamespaceSeries, seriesIteratorsPool, stepSize)
 		if err != nil {
 			return nil, err
 		}
 
 		// once we get the length of consolidatedSeriesBlocks, we can create a
 		// MultiSeriesBlocks list with the proper size
-		if multiNamespaceSeriesIdx == 0 {
+		if seriesIdx == 0 {
 			multiSeriesBlocks = make(MultiSeriesBlocks, len(consolidatedSeriesBlocks))
+			commonTags, err = storage.FromIdentTagIteratorToTags(multiNamespaceSeries[0].Tags)
+			if err != nil {
+				return nil, err
+			}
 		}
 
 		for consolidatedSeriesBlockIdx, consolidatedSeriesBlock := range consolidatedSeriesBlocks {
 			// we only want to set the start and end times once
-			if multiNamespaceSeriesIdx == 0 {
-				multiSeriesBlocks[consolidatedSeriesBlockIdx].Start = consolidatedSeriesBlock.Start
-				multiSeriesBlocks[consolidatedSeriesBlockIdx].End = consolidatedSeriesBlock.End
+			if seriesIdx == 0 {
+				multiSeriesBlocks[consolidatedSeriesBlockIdx].Metadata.Bounds.StepSize = consolidatedSeriesBlock.Metadata.Bounds.StepSize
```

Comment:
nit: can you make these names shorter? It's quite difficult to read at the moment

Also, can pull out multiSeriesBlocks[consolidatedSeriesBlockIdx].Metadata.Bounds into a var?